### PR TITLE
Added remove_content_xml flag and updated resource panel to use it

### LINKED ISSUE #761: Add remove_content_xml param to learning resource API
This param should remove the content_xml field from the information being returned if provided. It would enable better performance on the learning resource panel which currently may get megabytes of content_xml information before it can finish loading.

### PR DESCRIPTION
**Pull request by noisecapella**
Fixes #761

### CODE REVIEW

**Comment by giocalitri** [comment]
this needs to be rebased

File: rest/views.py [hunk 1]
```diff
@@ -550,6 +550,19 @@ def get_queryset(self):
         return LearningResource.objects.filter(
             id=self.kwargs['lr_id'])
 
+    def get(self, request, *args, **kwargs):
+        """Override get to filter on content_xml."""
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+
+        remove_content_xml = request.GET.get("remove_content_xml", None)
```
Comment (giocalitri) [inline rest/views.py]: you don't need to specify `None` here

**Comment by giocalitri** [comment]
Just two minor things, but what's the purpose of this PR?

**Comment by noisecapella** [comment]
It allows the resource panel to show up without downloading all of content_xml which is only used on the XML tab

**Comment by noisecapella** [comment]
Any other comments @giocalitri?

**Comment by giocalitri** [comment]
How to you get the XML then? Can you pull only that?

**Comment by noisecapella** [comment]
There are three AJAX calls made for the resource panel, one for each tab. The XML tab would call without the `remove_content_xml` flag which would bring in everything. Since `content_xml` is much bigger than any other field we don't need a flag for only `content_xml`

**Comment by giocalitri** [comment]
ok: I'll check functionality then

**Comment by giocalitri** [comment]
I noticed that when you open the panel there are 4 ajax requests right away:

1- for the metadata without the xml
2- for the static assets
3- for the vocabularies
4- for the  metadata with the xml

I honestly think this approach is wrong, especially with the introduction of this PR. 
IMHO the right approach should be to perform just the ajax request -1 and -3 and perform the others only if the user opens the related tab.

screen shot 2015-10-16 at 4 01 09 pm

**Comment by giocalitri** [comment]
other than the previous comment, this specific piece of functionality works fine.

**Comment by noisecapella** [comment]
I think it's out of the scope of this issue since it's behavior that existed before but feel free to file another issue for it

**Comment by giocalitri** [comment]
I actually do not think it is out of scope, because I do not understand why this PR is useful, then: is it because the first request is faster? But then we pay that with one more request for each panel...

**Comment by giocalitri** [comment]
I think I already talked about that kind of behavior several times, but I'll check and open an issue

**Comment by noisecapella** [comment]
It's useful because now it's only the XML tab which is blocked by the large AJAX request, the user can still interact with the other two panels in the meantime. It speeds up the user experience a little.

**Comment by noisecapella** [comment]
Since you're opening an issue for this, am I fine to merge this?

**Comment by giocalitri** [comment]
yep! go for it

**Pull request merged**